src/plugins/data/public/search/aggs/agg_types.ts

```diff
@@ -50,41 +52,44 @@ import { bucketAvgMetricAgg } from './metrics/bucket_avg';
 import { bucketMinMetricAgg } from './metrics/bucket_min';
 import { bucketMaxMetricAgg } from './metrics/bucket_max';
 
-export const aggTypes = {
-  metrics: [
-    countMetricAgg,
-    avgMetricAgg,
-    sumMetricAgg,
-    medianMetricAgg,
-    minMetricAgg,
-    maxMetricAgg,
-    stdDeviationMetricAgg,
-    cardinalityMetricAgg,
-    percentilesMetricAgg,
-    percentileRanksMetricAgg,
-    topHitMetricAgg,
-    derivativeMetricAgg,
-    cumulativeSumMetricAgg,
-    movingAvgMetricAgg,
-    serialDiffMetricAgg,
-    bucketAvgMetricAgg,
-    bucketSumMetricAgg,
-    bucketMinMetricAgg,
-    bucketMaxMetricAgg,
-    geoBoundsMetricAgg,
-    geoCentroidMetricAgg,
-  ],
-  buckets: [
-    dateHistogramBucketAgg,
-    histogramBucketAgg,
-    rangeBucketAgg,
-    dateRangeBucketAgg,
-    ipRangeBucketAgg,
-    termsBucketAgg,
-    filterBucketAgg,
-    filtersBucketAgg,
-    significantTermsBucketAgg,
-    geoHashBucketAgg,
-    geoTileBucketAgg,
-  ],
-};
+export function getAggTypes(deps: { uiSettings: IUiSettingsClient }) {
```

Comment:
Now that this has been wrapped with a provider pattern, we can (as a follow up step) get rid of the getters/setters that are currently being used to retrieve stateful services in each agg type, and instead explicitly pass those deps down.

Reply:
might make sense to pass down something that can be same on client and server (like getConfig function)